sdl-core/src/main/scala/io/smartdatalake/workflow/connection/jdbc/JdbcTableConnection.scala

```diff
@@ -64,9 +63,9 @@ case class JdbcTableConnection(override val id: ConnectionId,
                                db: Option[String] = None,
                                maxParallelConnections: Int = 1,
```

Comment:
Increasing to 3 should not do any harm because it is only a max value and the connections are created on demand. In most cases only one connection is used as you say, in this case no additional connection will be initialized. I will increase it to 3, it is a bit unintuitive to have a connection pool with capacity 1.